libp2p/transports/tcptransport.nim

```diff
@@ -12,262 +12,325 @@
 {.push raises: [].}
 
 import std/[sequtils]
-import stew/results
 import chronos, chronicles
-import transport,
-       ../errors,
-       ../wire,
-       ../multicodec,
-       ../connmanager,
-       ../multiaddress,
-       ../stream/connection,
-       ../stream/chronosstream,
-       ../upgrademngrs/upgrade,
-       ../utility
+import
+  ./transport,
+  ../wire,
+  ../multiaddress,
+  ../stream/connection,
+  ../stream/chronosstream,
+  ../upgrademngrs/upgrade,
+  ../utility
 
 logScope:
   topics = "libp2p tcptransport"
 
-export transport, results
+export transport, connection, upgrade
 
-const
-  TcpTransportTrackerName* = "libp2p.tcptransport"
+const TcpTransportTrackerName* = "libp2p.tcptransport"
 
 type
+  AcceptFuture = typeof(default(StreamServer).accept())
+
   TcpTransport* = ref object of Transport
     servers*: seq[StreamServer]
     clients: array[Direction, seq[StreamTransport]]
     flags: set[ServerFlags]
     clientFlags: set[SocketFlags]
-    acceptFuts: seq[Future[StreamTransport]]
+    acceptFuts: seq[AcceptFuture]
     connectionsTimeout: Duration
+    stopping: bool
 
   TcpTransportError* = object of transport.TransportError
 
-proc connHandler*(self: TcpTransport,
-                  client: StreamTransport,
-                  observedAddr: Opt[MultiAddress],
-                  dir: Direction): Future[Connection] {.async.} =
-
-  trace "Handling tcp connection", address = $observedAddr,
-                                   dir = $dir,
-                                   clients = self.clients[Direction.In].len +
-                                   self.clients[Direction.Out].len
+proc connHandler*(
```

Comment:
Do you think this should be public? Could you please add a doc to this proc? I don't understand what is going on here.

Reply:
Also not sure that's a descriptive name.

Reply:
this is a function that didn't really change - it's handling the interface between chronos and Connection, hence its name which I believe dates back to @dryajov's participation.